cfgov/v1/tests/test_links.py

```diff
@@ -22,6 +25,68 @@ def test_gov_link(self):
         self.assertIn('class="icon-link_text"', output)
 
 
+class LinkTest(TestCase):
+
+    def setUp(self):
+
+        self.page = mock.MagicMock()
+        self.request = mock.MagicMock()
+        self.request.url = 'http://localhost:8000/path/to/page/'
+        self.page.url = 'http://localhost:8000/path/to/some/other/page/'
+        self.page.live = True
+        self.page.specific.shared = True
+        self.context = mock.MagicMock()
+
+        def context_dict(name):
+            return {'request': self.request}[name]
+
+        self.context.__getitem__.side_effect = context_dict
+
+    def test_get_protected_url_no_page(self):
+        """
+        Confirm that in the absence of a page, get_protected_url returns the hash.
+        """
+        result = get_protected_url(self.context, None)
+        self.assertEqual(result, '#')
+
+    def test_get_protected_url_no_url(self):
+        """
+        Confirm that in the absence of a page URL, get_protected_url returns None.
+        """
+        self.page.url = None
+        result = get_protected_url(self.context, self.page)
+        self.assertIsNone(result)
+
+    def test_get_protected_url_live_page(self):
+        """
+        Confirm that if a live page is requested from the same host as the request
+        we get back the original page url.
+        """
+        self.page.url = 'http://localhost:8000/path/to/some/other/page/'
+        result = get_protected_url(self.context, self.page)
+        self.assertEqual(result, self.page.url)
+
+    def test_get_protected_url_non_live_page(self):
+        """
+        Confirm that if a non-live page is request, we get back the hash.
```

Comment:
> I don’t know what “the absence of a page” might mean. Also not sure what returning “the hash” means as a behavior?

"Absence of a page" means "the page object is None." Sorry, should have been clearer on that.

> Does this mean that any link would be null without a URL? If yes, +1

Yes.

> The reference to “original page url" throws me, but if this means that when a page is requested from www, we get back the requested page at www, then +1

Yes, this is what that means. 

> I think this is confirming that a staged page cannot be requested from www, which is good. Again, not sure what returning the hash means as a behavior.

Again, I should have been clearer on this. This function is actually a template tag that "protects" a URL so that, as you say, a staged URL cannot be requested from www. The return value of that function is the hash, '#', which is interpolated into the template, which is what I meant.

> I think this means that if I’m on staging.cf.gov/about-us/ and I request www.cf.gov/about-us/blog for whatever reason, I get back staging.cf.gov/about-us/blog. If yes, +1

Correct.